### PR TITLE
Vijayasree/Refactor forms ts

### PR DESCRIPTION
**Pull request by vijayasree-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/form/input.tsx [hunk 1]
```diff
@@ -1,15 +1,63 @@
...
+    showLabel?: string
+    label_focus_color?: string
+    background?: string
+    label?: string
+    error?: string
+}
+
+type ValidType = {
+    background?: string
+}
```
Comment (prince-deriv) [inline src/components/form/input.tsx:46]: All used as a props. Rename from `Type` ending to `Props` ending

**Review by prince-deriv — APPROVED** [review]

File: src/components/form/input.tsx [hunk 2]
```diff
@@ -1,15 +1,63 @@
...
+    handleError?: (urrent_input: React.MutableRefObject<HTMLInputElement>) => void
+    height?: string
+    id?: string
+    label?: string
+    label_color?: string
+    label_hover_color?: string
+    maxLength?: number
+    tablet_background?: string
+    width?: string
+}
```
[the remark this answers is not in the document]
Comment (sean-binary) [reply src/components/form/input.tsx:24]: Same for the button props.

**Review by NikitK-deriv — APPROVED** [review]
After resolving Sean's comment LGTM 👍

**Review by sean-binary — DISMISSED** [review]

**Review by sean-binary — APPROVED** [review]

File: src/components/form/button.ts [hunk 3]
```diff
@@ -1,9 +1,25 @@
 import styled, { css } from 'styled-components'
-import PropTypes from 'prop-types'
 import { Paddings, Margins } from 'themes/function'
 import device from 'themes/device.js'
 
-export const SharedButtonStyle = css`
+interface ButtonProps extends React.ComponentPropsWithoutRef<'button'>, SharedButtonStyleProps {
```
Comment (mohammad-rework) [inline src/components/form/button.ts:5]: Since our convention is using `type` instead of `interface` let's use `type` as much as possible

**Review by prince-deriv — APPROVED** [review]

**Pull request merged**